invitev2.go

```diff
@@ -0,0 +1,125 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/tidwall/gjson"
+)
+
+// InviteV2Request and InviteV2StrippedState are defined in
+// https://matrix.org/docs/spec/server_server/r0.1.3#put-matrix-federation-v2-invite-roomid-eventid
+
+func NewInviteV2Request(event *HeaderedEvent, state []InviteV2StrippedState) (
+	request InviteV2Request, err error,
+) {
+	if event.RoomVersion == "" {
+		err = errors.New("gomatrixserverlib: malformed headered event")
```

Comment:
Say what failed e.g "missing room version". For now people can infer since we just have that as data but if and when we add more fields this message becomes unhelpful.

Reply:
Fixed.